ignite/metrics/epoch_metric.py

```diff
@@ -80,7 +80,11 @@ def update(self, output):
                 self.compute_fn(self._predictions, self._targets)
             except Exception as e:
                 warnings.warn("Probably, there can be a problem with `compute_fn`:\n {}.".format(e),
-                              RuntimeWarning)
+                              EpochMetricWarning)
 
     def compute(self):
         return self.compute_fn(self._predictions, self._targets)
+
+
+class EpochMetricWarning(UserWarning):
```

Comment:
I think, this class should be moved to ignite/exceptions.py

Reply:
I thought about this. This warning is very specific to `EpochMetric` and it is not an exception...

Reply:
Currently it may be specific to `EpochMetric`. But what if we just rename it to e.g. `EpochWarning` and slightly change the text associated with this warning to make it more general (so that we could for example reuse it in the `Engine`?

(And btw: every warning is also an exception: https://docs.python.org/3/library/warnings.html#warning-categories)

Reply:
Currently it may be specific to `EpochMetric`. But what if we just rename it to e.g. `EpochWarning` and slightly change the text associated with this warning to make it more general (so that we could for example reuse it in the `Engine`?

(And btw: every warning is also an exception: https://docs.python.org/3/library/warnings.html#warning-categories)

Reply:
> And btw: every warning is also an exception: https://docs.python.org/3/library/warnings.html#warning-categories

I see :) Thanks 

Well, here it is introduced to be able to filter it as the issue states that. So reusing this warning can be later misleading, IMO.

Reply:
From our discussion somewhere else: 
> For EpochMetricWarning is just to ignore a sanity check as it is annoying in some cases as issues stated. Reusing is not a good idea I think